scrapy/exporters.py

```diff
@@ -121,13 +128,9 @@ def finish_exporting(self):
         self.file.write(b"]")
 
     def export_item(self, item):
-        if self.first_item:
-            self.first_item = False
-        else:
-            self.file.write(b',')
-            self._beautify_newline()
         itemdict = dict(self._get_serialized_fields(item))
         data = self.encoder.encode(itemdict)
+        self._add_comma_after_first()
         self.file.write(to_bytes(data, self.encoding))
```

Comment:
If we want to be super-strict on it, to_bytes can also fail.

Reply:
It was harder to write a test case for encoding that worked across versions. :) In the end the sanest thing was a new test class, sorry for the bloat.